libraries/abstractions/common_io/test/test_iot_spi.c

```diff
@@ -1515,8 +1453,34 @@ TEST( TEST_IOT_SPI, AFQP_IotSPI_CancelFuzzing )
         TEST_ASSERT_EQUAL( IOT_SPI_NOTHING_TO_CANCEL, lRetVal );
     }
 
-    /* Close SPI handle */
     lRetVal = iot_spi_close( xSPIHandle );
     TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
 }
 /*-----------------------------------------------------------*/
+
+
+
+/*-----------------------------------------------------------*/
+
+static void prvAppendToMessage( size_t * pOffset,
+                                uint8_t * pBuffer,
+                                size_t bufferLen )
+{
+    size_t i = 0;
+    size_t offset = *pOffset;
+
+    for( ; i < bufferLen && offset + 2 < _MESSAGE_LENGTH; i++ )
```

Comment:
What's the relationship between bufferLen and _MESSAGE_LENGTH?

Reply:
This function append all the data of an input buffer to _cMsg buffer. The bufferLen is the length of input buffer and _MESSAGE_LENGTH is length of _cMsg.

Reply:
It would seem safer to use _MESSAGE_LENGTH only in the call site(s). Then modify this routine to take the various buffer sizes as explicit input variables on the stack.